sanitizers/detectors/sql_injection.go

```diff
@@ -18,31 +20,33 @@ var syntaxErrors = []*regexp.Regexp{
 	regexp.MustCompile(`\S+ ERROR: syntax error at or near .* \(SQLSTATE 42601\)`),         // PostgreSQL error message
 }
 
-type SQLInjection struct {
-	id    int    // numeric identifier to distinguish between the detectors for the various call sites
-	query string // the SQL query that has been executed
-	err   error  // the error resulting from executing the SQL query
+var _ Detector = (*sqlInjection)(nil)
+
+type sqlInjection struct {
+	id    int
+	cmd   string
+	err   error
+	vargs []any
 }
 
-// Make sure that the SQL injection detector implements the Detector interface
-var _ Detector = (*SQLInjection)(nil)
+var SQLInjection = sqlInjection{}
 
-func (si *SQLInjection) Detect() error {
-	if isSyntaxError(si.err) {
-		return SQLInjectionError
+func (sqli *sqlInjection) Detect() {
+	if isSyntaxError(sqli.err) {
+		if len(sqli.cmd) > 0 {
+			reporter.ReportFindingf("%s: query %s, args [%s]", SQLInjectionError.Error(), sqli.cmd, fmt.Sprint(sqli.vargs...))
+		} else {
+			reporter.ReportFindingf("%s: args [%s]", SQLInjectionError.Error(), fmt.Sprint(sqli.vargs...))
+		}
+		return
```

Comment:
The `return` statement is not needed here since we panic when reporting the finding.

Reply:
True